src/control.c

```diff
@@ -821,9 +828,15 @@ handle_multi_dev(int argc, char *argv[])
 	else
 		file_count = nr_active_devices;
 
-	for (i = 0; i < file_count; i++) {
-		args.path = argv[i + 2];
-		args.id = i + 1;
+	if (pthread_mutex_init(&device_mutex, NULL)) {
+		printf(ERR "Failed to initialize device mutex\n");
+		return;
+	}
+
+	for (id = 0; id < file_count; id++) {
+		pthread_mutex_lock(&device_mutex);
+		args.path = argv[id + 2];
+		args.id = id + 1;
 
 		ret = pthread_create(&threads[i], NULL, start_device,
```

Comment:
This should be the line causing the trouble i think. You are using 'i' instead of 'id' for thread creation, but are closing the right amount of threads

Reply:
It's strange. I tested the code with 3 attenuators without any particular issue. Let me check the code...probably it's a merge issue (I'll look for some typo).

Reply:
Ok, I see the problem. In my local code I'm using 'i' inside the for instead of 'id'. So the index used inside the for should be the same of the one used for the thread array. 

The following line should be enough to solve the issue.
ret = pthread_create(&threads[id], NULL, start_device,  (void *)&args);

Sorry about that but I extended my fork with a lot of features so I merged the code manually on GitHub to perform the pull request.
Please fix the typo and merge the code.
Most probably I will upload other pull requests in the next days with the other features.
Let me know if you need more specifics.
Thanks!

Reply:
Can you push this minor change to this branch please? Or @thuehn can you add this change, as i have not the right to do so.

Reply:
I've just pushed this minor change on my branch. Let me know if this solve the issue or if there is any other problem to solve.

Reply:
@ikstream yes, you are right. I've been to busy at work and I wasn't able to find the time to push the code. I'll do that as soon as possible and I'll keep you posted. Thanks!

Reply:
Hello Guys. 
@ikstream I finally found the time to commit the new features I was talking about some time ago. You'll find that in pull request  #14.
Have a nice day!